### ISSUE TITLE
Ability to either suffix or prefix the staging environment

### ISSUE THREAD

**Issue by mpuittinen** [SamVerschueren/aws-lambda-env#2]
In order to have lambda functions grouped logically by application, we are planning to use the notation FunctionName_[version]_[stage] for our functions. To make the module more versatile, the proposal is to make aws-lambda-env return  methods getStagePrefix / getStageSuffix instead of returning the outcome directly. This would also allow for extending the method with other functions if required. 
If that is OK, I can implement the change to and submit a pull request.

**Comment by SamVerschueren** [comment]
Hi @mpuittinen 

Thanks for the proposal. So you want something like this:

``` javascript
require('aws-lambda-env')();
// staging_HelloWorld -> staging

require('aws-lambda-env')(':');
// staging:HelloWorld -> staging

require('aws-lambda-env').suffix();
// HelloWorld_staging -> staging

require('aws-lambda-env').suffix(':');
// HelloWorld:staging -> staging

require('aws-lambda-env').prefix();
// staging_HelloWorld -> staging

require('aws-lambda-env').prefix(':');
// staging:HelloWorld -> staging
```

**Comment by mpuittinen** [comment]
Hi Sam!

Ues, that is what I had in mind (or actually only the last 4 examples). The first one can be difficult to achive since it would return the stage string while at the same time, examples 3-6 expect that to be a collection with methods prefix and suffix. The most consistent implementation would be to only have the 4 last ones (which would unfortunately require to refactor some existing code that rely on the current implementation).

-Mikael

**Comment by SamVerschueren** [comment]
It's not that hard to achieve :). Something like this will work for example:

``` javascript
module.exports = function() {};
module.exports.suffix = function() {};
module.exports.prefix = module.exports;
```

**Comment by mpuittinen** [comment]
Yes, it actually seems to work. Which led me to another (in my opinion more elegant) approach:

```
var lambdaenv = require(‘aws_lambda_env’)({ stageNaming: ‘suffix’ });
var stage = lambdaenv.stage;
```

i.e. in case the input is a collection with the parameters to use, module.exports returns a collection with the parsed environment parameters. (more parameters / functions can be added over time

I’ll create a fork and submit a pull request once it’s implemented / tested and documented. I’ll also add name() as a function that returns the function name itself (hiding the environment variables in case they change over time).

Cheers,
    Mikael

**Comment by SamVerschueren** [comment]
Hi @mpuittinen 

Thanks for the suggestions. Would you mind letting me think this through first? I want to go over all the options first.

You can off course start your implementation and adjust later on if necessary.

**Comment by mpuittinen** [comment]
Actually, I think I’ll implement my own module after all, I’d prefer not to overload the “main module” with a function that returns a value in some cases and the keep the module generic for returning AWS Lambda function info (name, memory size, stage based on name, version based on name, etc…). i.e, have require(‘some-module.js’) to always return a collection / object that contains attributes / functions.

Cheers,
    Mikael

**Comment by SamVerschueren** [comment]
Well I like the idea of being able to retrieve the environment at the end of the name as well (like in my first suggestions). Maybe not really the name of the function because it's not the goal of the module.

So maybe I (or someone) can still implement the first suggestion and then you can use this module in your more broader module?

**Comment by SamVerschueren** [comment]
Implemented in `v1.1.0`.

**Comment by mpuittinen** [comment]
Hi!

Thanks for the change. I already published my alternative module, I think I’ll stick to that.

https://www.npmjs.com/package/aws-lambda-info https://www.npmjs.com/package/aws-lambda-info

Cheers, 

Mikael
